### ISSUE TITLE
Plotter segfaults in for loop

### ISSUE THREAD

**Issue by janbechtold** [pyvista/pyvista#3869]
### Describe the bug, what's wrong, and what you expected.

I am iterating over meshes to show them but it throws a segfault after the second visualization.
Expected behaviour: use .plot() in for loop with new window every time.

### Steps to reproduce the bug.

```python
import pyvista
for i in range(3):
    pyvista.Sphere().plot()
```
close the window with `q` or the cross. Second plotting segfaults.

### System Information

```shell
Python 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0]

           pyvista : 0.37.0
               vtk : 9.2.5
             numpy : 1.20.2
           imageio : 2.24.0
            scooby : 0.7.0
             pooch : v1.6.0
        matplotlib : 3.6.3
             scipy : 1.6.2
              tqdm : 4.62.3
            meshio : 4.4.6
```

**Comment by banesullivan** [comment]
The other parts of the PyVista report are actually quite important. Would you please paste the full report from:

```py
import pyvista as pv
print(pv.Report())
```

**Comment by janbechtold** [comment]
Sorry, I missed the upper part when copying. I updated the original post.

**Comment by banesullivan** [comment]
Thanks! this does remind me of #558... this is quite odd. I'm not sure if I'll be able to reproduce this but I'll try to look into it.

> close the window with q or the cross. Second plotting segfaults.

This happens on the second plot *always*? It concerns me that this is happening even with the `q` key...

-----

Does this machine have a dedicated GPU? I'm thrown off by the 36-core CPU and using Mesa for graphics... In my experience, 36-cores are quite a lot (desktop machine level, right?), and thus indicate a likely presence of a dedicated graphics card. I wonder if something is off with the graphics configuration.

**Comment by janbechtold** [comment]
Yes, I've never gotten to see the third mesh / window.
There is a dedicated graphics card, but it's dedicated for other processes. So I cannot use it. Therefore using Mesa for graphics looks like the correct configuration.

Do you think it is this setup that causes the segfault? Were you able to reproduce it?

Is there maybe another way for me to achieve it, like reusing the plotter instead of closing it (and trigger loading the next mesh with another key)?

**Comment by banesullivan** [comment]
I'm unable to reproduce this... I'd like to set up a pure VTK script for you to try to see if you can reproduce the issue with just that. That would at least help rule out anything specific in PyVista going wrong. Could you give this a try:

```py
import vtk
source = vtk.vtkConeSource()
mapper = vtk.vtkPolyDataMapper()
output_port = source.GetOutputPort()
mapper.SetInputConnection(output_port)
actor = vtk.vtkActor()
actor.SetMapper(mapper)

for i in range(3):
    print(i)
    ren = vtk.vtkRenderer()
    renWin = vtk.vtkRenderWindow()
    renWin.AddRenderer(ren)
    iren = vtk.vtkRenderWindowInteractor()
    iren.SetRenderWindow(renWin)
    ren.AddActor(actor)
    iren.Initialize()
    renWin.Render()
    iren.Start()
```

**Comment by janbechtold** [comment]
Good idea. Thanks for the script. It works. No segfault (core dumped) thrown.

I need to clarify failing on the second plot. I realized I cannot rotate the mesh in the second plot. It appears frozen and the segfault occurs after a short time.

I went back to issue #558. When I close the plot with `cross` or `q` a black window remains.
![grafik](https://user-images.githubusercontent.com/22617469/214791246-0f7303e1-ddf3-4b32-8aab-4e625d0a57eb.png)
`exit()` however does not throw a segfault.

**Comment by dcbr** [comment]
This makes me think of the bug I encountered on Windows due to the `TerminateApp` call (see [vtk!8696](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8696) for the fix and link to issue). Just to be sure this is not the cause, can you add the line `iren.TerminateApp()` after `iren.Start()` (still within the for loop) and see if there are still no issues then?

**Comment by janbechtold** [comment]
Still no issues.

So currently I don't know what else to try.
But the issue is also not too pressing on my end.
I can still go back to open3d, which works in a loop.

**Comment by banesullivan** [comment]
@janbechtold, I see you have VTK 9.2.2. Would you be able to try VTK 9.1 and VTK 9.2.5 to issue if the issue is persistent with PyVista across those versions?

**Comment by banesullivan** [comment]
Okay, multiple people seeing this concerns me. The only commonality seems to be using Mesa?

Is either of you willing to install previous versions of PyVista and VTK to find a version where this was working? Then perhaps git bisecting? I know it's a big ask, but it's tough for me to track this down without being able to reproduce myself.

**Comment by ryan-salo** [comment]
I downgraded to `vtk==9.1` and the issue went away with both versions of pyvista `0.36.1` and `0.38.dev0`.

**Comment by banesullivan** [comment]
Aha, thank you. This really helps! @akaszynski, I'm curious if you can reproduce as I suspect this may be tied to https://github.com/pyvista/pyvista/issues/1096#issuecomment-1374000281

**Comment by bjlittle** [comment]
@banesullivan I'm seeing this issue also and the common factor is `vtk` v9.2.5. Everything is fine with v9.2.2

I'm getting segfaults whenever I plot in a loop, the second plot always segfaults.

**Comment by bjlittle** [comment]
@banesullivan Don't know if this issue is related to the destruction of one render window and the creation of the next...

If you want me to try anything at this end, then please advise and I'll try anything you suggest that might help diagnose the root cause :+1:

**Comment by bjlittle** [comment]
This also appears to be the case with `vtk` 9.2.4 when installed from PyPI.

**Comment by banesullivan** [comment]
9.2.3-4 are the same as 9.2.5 (there was a minor bug in 9.2.5 that we patched in 9.2.5). Actually 9.2.3-4 are being yanked from PyPI

**Comment by banesullivan** [comment]
@bjlittle, thanks for also reporting on this! I'll see what I can do to continue to look into this

**Comment by bjlittle** [comment]
> 9.2.3-4 are the same as 9.2.5 (there was a minor bug in 9.2.5 that we patched in 9.2.5). Actually 9.2.3-4 are being yanked from PyPI

Awesome, thanks for clarifying @banesullivan. Well, that said, it makes sense with the behaviour that I'm seeing.

Suggests that the change in `vtk` that's causing this issue was introduced in 9.2.3-4.

I've not been able to test with 9.2.3... it's not available on conda-forge and I guess has already been yanked on PyPI.

**Comment by antonagafonov** [comment]
Hi,
I have the same issue,
After the second plot of pyvista sometimes I have :
`Segmentation fault (core dumped)`

Info:
```

--------------------------------------------------------------------------------
  Date: Sun Feb 05 16:55:18 2023 IST

                OS : Linux
            CPU(s) : 20
           Machine : x86_64
      Architecture : 64bit
               RAM : 62.6 GiB
       Environment : Python
       File system : ext4
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA RTX A6000/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 520.61.05

  Python 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 15:55:03)
  [GCC 10.4.0]

           pyvista : 0.37.0
               vtk : 9.2.5
             numpy : 1.24.1
           imageio : 2.25.0
            scooby : 0.7.1
             pooch : v1.6.0
        matplotlib : 3.6.3
           IPython : 8.8.0
             scipy : 1.10.0
              tqdm : 4.64.1
            meshio : 5.3.4
         pythreejs : 2.4.1
```

**Comment by banesullivan** [comment]
I'm pretty sure that this was tied to the X11 close path on older Mesa and has since resolved itself.
